pkg/webui/console/containers/device-onboarding-form/device-type-form-section/device-type-repository-form-section/device-selection/band-select/index.js

```diff
@@ -12,9 +12,64 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import connect from './connect'
-import BandSelect from './band-select'
+import React from 'react'
+import { defineMessages } from 'react-intl'
+import { useSelector } from 'react-redux'
 
-const ConnectedBandSelect = connect(BandSelect)
+import Field from '@ttn-lw/components/form/field'
+import Select from '@ttn-lw/components/select'
+import { useFormContext } from '@ttn-lw/components/form'
 
-export { ConnectedBandSelect as default, BandSelect }
+import PropTypes from '@ttn-lw/lib/prop-types'
+
+import { selectDeviceModelFirmwareVersions } from '@console/store/selectors/device-repository'
+
+import { SELECT_OTHER_OPTION } from '../../../../utils'
+import messages from '../../../../messages'
+
+const m = defineMessages({
+  title: 'Profile (Region)',
+})
+
+const formatOptions = (profiles = []) =>
+  profiles
+    .map(profile => ({
+      value: profile,
+      label: profile,
+    }))
+    .concat([{ value: SELECT_OTHER_OPTION, label: messages.otherOption }])
+
+const BandSelect = props => {
```

Comment:
For this component I refactored it to use just a single module and use `useSelector`-hooks directly. I think this makes the component a lot easier to grasp.

Reply:
I will those in the provisioning part as well